src/core/reduce.rkt

```diff
@@ -24,20 +25,23 @@
     [`(lambda ,vars ,body)
      `(λ ,vars ,(simplify body))]
     [`(neg ,arg)
-      (define op* (car (get-parametric-operator '- '(binary64))))
+      (define op* (get-parametric-operator '- (representation-name (*output-repr*))))
       (define arg* (simplify arg))
       (define val (eval-application op* arg*))
       (or val (simplify-node (list 'neg arg*)))]
-    [`(,(? (curry hash-has-key? parametric-operators) op) ,args ...)
-     ; Get the parameterized op for binary64
-     ; Correct arg length is taken from get-operator-argc and not 'args since these exprs
-     ; are v-ary while get-operator-argc is not
-     (define arg-len (length (get-operator-argc op)))
-     (define op* (car (get-parametric-operator op (make-list arg-len 'binary64))))
+    [(list (? (λ (x) (regexp-match? #rx"[A-Za-z0-9_]+(->)[A-Za-z0-9_]+" 
+                                        (symbol->string x))) op)
+               body) ; conversion (e.g. posit16->f64)
+      expr] ; TODO: figure out what to do here
+    [`(,op ,args ...)
+     ; Correct arg-length is taken from get-operator-itype since some operators are allowed
+     ; to be v-ary
+     (define argc (length (get-operator-itype op)))
+     (define op* (apply (curry get-parametric-operator op) 
+                        (make-list argc (representation-name (*output-repr*)))))
```

Comment:
You don't need the `curry` here. The signature of `apply` is `(apply fn arg1 arg2 ... list-of-remaining-args)`.

Reply:
And wait: which operators are variary? I think the comment is not right.